src/core/Akka/IO/TcpConnection.cs

```diff
@@ -646,51 +685,34 @@ protected override void PostRestart(Exception reason)
             throw new IllegalStateException("Restarting not supported for connection actors.");
         }
 
-        private PendingWrite CreatePendingWrite(IActorRef commander, WriteCommand write, ConnectionInfo info)
+        private Option<PendingWrite> GetNextWrite(IEnumerable<(SimpleWriteCommand Command, IActorRef Sender)> headCommands = null)
         {
-            var head = write;
-            WriteCommand tail = Write.Empty;
-            while (true)
+            headCommands = headCommands ?? new List<(SimpleWriteCommand Command, IActorRef Sender)>();
```

Comment:
If there's a way we can use a static `List<(SimpleWriteCommand Command, IActorRef Sender)>.Empty` or something here instead of `new`-ing up a list each time, I think that would help reduce allocations here.

Reply:
`ImmutableList<(SimpleWriteCommand Command, IActorRef Sender)>.Empty` would do it.

Reply:
Good point, thanks!